src/mocks/TimelineMock.ts

```diff
@@ -0,0 +1,257 @@
+            }
+        },
+        "The previous batch from a sync returns the previous events": assert => {
+            const timeline = new TimelineMock(SENDER);
+            timeline.append(20);
+            const sync = timeline.sync(undefined, 10);
+            const messages = timeline.messages(sync.timeline.prev_batch, undefined, "b");
+            const events = messages.chunk.map(e => e.event_id).reverse();
+            assert.deepEqual(events, eventIds(0, 10));
+        },
+        "Two consecutive message fetches are continuous if no new events are inserted": assert => {
+            const timeline = new TimelineMock(SENDER);
+            timeline.append(30);
+            const sync = timeline.sync(undefined, 10);
+            const messages1 = timeline.messages(sync.timeline.prev_batch, undefined, "b");
+            const events1 = messages1.chunk.map(e => e.event_id).reverse();
+            const messages2 = timeline.messages(messages1.end, undefined, "b");
+            const events2 = messages2.chunk.map(e => e.event_id).reverse();
+            assert.deepEqual(events1, eventIds(10, 20));
+            assert.deepEqual(events2, eventIds(0, 10));
+        },
+        "Two consecutive message fetches detect newly inserted event": assert => {
+            const timeline = new TimelineMock(SENDER);
+            timeline.append(30);
+            const messages1 = timeline.messages(eventId(20), undefined, "b", 10);
+            const events1 = messages1.chunk.map(e => e.event_id).reverse();
+            timeline.insertAfter(eventId(9), 1);
+            const messages2 = timeline.messages(eventId(10), undefined, "b", 10);
+            const events2 = messages2.chunk.map(e => e.event_id).reverse();
+            assert.deepEqual(events1, eventIds(10, 20));
+            const expectedEvents2 = eventIds(1, 10);
+            expectedEvents2.push(eventId(30));
+            assert.deepEqual(events2, expectedEvents2);
+        },
+        "A sync that receives no events has the same next batch as it started with": assert => {
+            const timeline = new TimelineMock(SENDER);
+            timeline.append(10);
+            const sync1 = timeline.sync();
+            const sync2 = timeline.sync(sync1.next_batch);
+            assert.equal(sync1.next_batch, sync2.next_batch);
+        }, 
+        "An event inserted in the midle still shows up in a sync": assert => {
+            const timeline = new TimelineMock(SENDER);
+            timeline.append(30);
+            const sync1 = timeline.sync(undefined, 10);
+            const sync2 = timeline.sync(sync1.next_batch, 10)
+            assert.deepEqual(sync2.timeline.events, []);
+            assert.equal(sync2.timeline.limited, false);
+            timeline.insertAfter(TIMELINE_START_TOKEN, 1);
+            const sync3 = timeline.sync(sync2.next_batch, 10)
+            const events = sync3.timeline.events.map(e => e.event_id);
+            assert.deepEqual(events, eventIds(30, 31));
```

Comment:
I would have expected just event `31` in the last sync?

Reply:
oh, the lower bound is not included, this might be clearer:
```suggestion
            assert.deepEqual(events, [eventId(31)]);
```